tsdb/chunks/head_chunks.go

```diff
@@ -0,0 +1,711 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chunks
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	tsdb_errors "github.com/prometheus/prometheus/tsdb/errors"
+	"github.com/prometheus/prometheus/tsdb/fileutil"
+)
+
+// Segment header fields constants.
+const (
+	// MagicHeadChunks is 4 bytes at the head of a segment file.
+	MagicHeadChunks = 0x0130BC91
+
+	headChunksFormatV1 = 1
+	inBufferShards     = 16
+	writeBufferSize    = 2 * 1024 * 1024
+)
+
+var (
+	// DefaultHeadChunkSegmentTime is the default chunks segment time range.
```

Comment:
time range?

Can we be explicit here? (: I think it's the segmentMaxTime kind of? as it's a threshold when we start to create segments, no?

Reply:
Yeah. I will modify the name.